lib/components/user/monitored-trip/trip-status.js

```diff
@@ -0,0 +1,391 @@
+    }
+  }
+
+  renderHeader = () => {
+    const { monitoredTrip } = this.props
+    const journeyState = this.getJourneyState()
+
+    let headerText
+    if (!monitoredTrip.isActive) {
+      headerText = 'Trip monitoring is paused'
+    } else if (monitoredTrip.snoozed) {
+      headerText = 'Trip monitoring is snoozed for today'
+    } else if (journeyState.tripStatus === 'NO_LONGER_POSSIBLE') {
+      headerText = 'Trip is no longer possible.'
+    } else if (journeyState.tripStatus === 'NEXT_TRIP_NOT_POSSIBLE') {
+      headerText = 'Trip is not possible today.'
+    } else if (
+      journeyState.tripStatus === 'TRIP_UPCOMING' ||
+        journeyState.tripStatus === 'TRIP_ACTIVE'
+    ) {
+      headerText = this.renderRealtimeInfo().headerText
+    } else {
+      // trip status not yet calculated, return null
+      return null
+    }
+
+    const secondsSinceLastCheck = moment().diff(
+      moment(journeyState.lastCheckedEpochMillis),
+      'seconds'
+    )
+    const lastCheckedText = `Last checked: ${formatDuration(secondsSinceLastCheck)} ago`
+
+    return (
+      <Panel.Heading>
+        <h3>{headerText}</h3>
+        <h6>{lastCheckedText}</h6>
+      </Panel.Heading>
+    )
+  }
+
+  renderBody = () => {
+    const journeyState = this.getJourneyState()
+    const { monitoredTrip } = this.props
+
+    let body
+    if (!monitoredTrip.isActive) {
+      body = 'Resume trip monitoring to see the updated status'
+    } else if (monitoredTrip.snoozed) {
+      body = 'Unsnooze trip monitoring to see the updated status'
+    } else if (journeyState.tripStatus === 'NO_LONGER_POSSIBLE') {
+      body = 'The trip planner was unable to find your trip on any selected days of the week. Please try re-planning your itinerary to find an alternative route.'
+    } else if (journeyState.tripStatus === 'NEXT_TRIP_NOT_POSSIBLE') {
+      body = 'The trip planner was unable to find your trip today. Please try re-planning your itinerary to find an alternative route.'
+    } else if (
+      journeyState.tripStatus === 'TRIP_UPCOMING' ||
+        journeyState.tripStatus === 'TRIP_ACTIVE'
+    ) {
+      body = this.renderRealtimeInfo().body
+    } else {
+      // trip status not yet calculated, return null
+      return null
+    }
+
+    return (
+      <Panel.Body>{body}</Panel.Body>
+    )
+  }
+
+  renderAlerts = () => {
+    const { longDateFormat, monitoredTrip, timeFormat } = this.props
+    const journeyState = this.getJourneyState()
+    const { matchingItinerary } = journeyState
+
+    // don't show alerts for trips that are no longer possible, not possible for
+    // the current trip or if the trip is paused or snoozed
+    if (
+      !monitoredTrip.isActive ||
+        monitoredTrip.snoozed ||
+        journeyState.tripStatus === 'NO_LONGER_POSSIBLE' ||
+        journeyState.tripStatus === 'NEXT_TRIP_NOT_POSSIBLE'
+    ) {
+      return null
+    }
+
+    // don't show alerts for upcoming trips that aren't within the lead time
```

Comment:
No, because since they are in upcoming trips, it is possible that these alerts are now stale due to the trip not having been calculated in a while. Therefore, they shouldn't be shown until after the lead time. I added a comment about this in 7d5505703026f41c849998ada8d6a66278f321b5